docs/upgrading.md

```diff
@@ -53,6 +55,79 @@
     - [Removing unnecessary metadata using the resourceprocessor](#removing-unnecessary-metadata-using-the-resourceprocessor)
     - [Moving record-level attributes used for metadata to the resource level](#moving-record-level-attributes-used-for-metadata-to-the-resource-level)
 
+## Upgrading to v0.103.0-sumo-0
+
+### `sumologic` configuration: modified the `configuration files` merge behaviour
+
+Modified the configuration merge behaviour to perform overwrite instead of update for `collector_fields` key of [Sumo Logic Extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/sumologicextension#configuration).
+
+For example, if two configuration files(say `conf1.yaml` and `conf2.yaml`) define the `collector_fields`,
+
+In previous flow, the values of `collector_fields` from `conf2.yaml` will be added to `conf1.yaml` and the effective configuration will have vaules from both the configurations.
+
+Starting from `v0.103.0-sumo-0`, the values of `collector_fields` tag from `conf1.yaml` will be overwritten by values from `conf2.yaml` and the effective configuration will have `collector_fields` value of `conf2.yaml` only.
+
+For example:
+
+`conf1.yaml`:
+
+```
+extensions:
+  sumologic:
+    collector_description: "My OpenTelemetry Collector"
+    collector_fields:
+      cluster: "cluster-1"
+    some_list:
+      - element 1
+      - element 2
+```
+
+`conf2.yaml`:
+
+```
+extensions:
+  sumologic:
+    collector_fields:
+      zone: "eu"
+    some_list:
+      - element 3
+      - element 4
+```
+
+effective configuration (`old behaviour`)
+
+```
+extensions:
+  sumologic:
+    collector_description: "My OpenTelemetry Collector"
+    collector_fields:
+      cluster: "cluster-1"
+      zone: "eu"
+    some_list:
+      - element 3
+      - element 4
+```
+
+effective configuration (`new behaviour`)
```

Comment:
is there an issue describing the need for the new behavior? If so please link the issue to the PR. If its a Jira issue then let's maybe describe the requirement in the PR.

Reply:
https://docs.google.com/document/d/1U9yb2SAW3Q-c3zWimMwoFRAP0jsdGioijlbcUCHZ4wY

This change was done as part of this feature.